src/commands/upload.rs

```diff
@@ -256,32 +256,55 @@ pub struct UpdateSubmissionMutation;
 )]
 pub struct ValidateSubmissionMutation;
 
-async fn upload_file(
-    client: &reqwest::Client,
-    file: impl AsRef<Path>,
-    upload_url: &Url,
-    content_type: &str,
-) -> Result<()> {
-    let file = tokio::fs::File::open(file).await?;
-    let content_len = file.metadata().await?.len();
-    let response = client
-        .put(upload_url.to_string())
-        .header(CONTENT_LENGTH, content_len)
-        .header(CONTENT_TYPE, content_type)
-        .body(file)
-        .send()
-        .await?;
-    if !response.status().is_success() {
+fn get_use_case_file(
+    files: &[update_use_case_mutation::UpdateUseCaseMutationCreateUseCaseVersionNodeFiles],
+    kind: update_use_case_mutation::ProjectVersionFileKind,
+) -> Result<(Id, &Url)> {
+    if let Some(file) = files
+        .iter()
+        .find(|f| f.kind == kind)
+        .and_then(|f| f.upload_url.as_ref().map(|url| (&f.id, url)))
+        .map(|(id, url)| Id::parse_node_id(id).map(|id| (id, url)))
```

Comment:
yeah, its not ideal, but I don't think your suggestion would work because I want a `Result<Option<(Id, Url)>>` in the end. I'll see if I can rework it though